src/averageAges.js

```diff
@@ -37,7 +43,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter((person) => person.sex === 'f');
+
+  women = withChildren
+    ? women.filter((mother) => people.some((person) => (
+      mother.name === person.mother
+    )))
+    : women;
+
+  const sumOfAge = women.reduce((prev, woman) =>
+    woman.died - woman.born + prev, 0);
```

Comment:
don't repeat yourself: create separate function for calculating average age for people array